### ISSUE TITLE
TCP packets only printing ack flags regardless of what flag maybe present

### ISSUE THREAD

**Issue by AlienX2001** [seladb/PcapPlusPlus#896]
Hey, its me again.
The screenshot for the issue
![image](https://user-images.githubusercontent.com/64488123/171049656-c3feb9a1-4f1e-48b2-83a5-a5d0f05cc845.png)

The left side is pcapplusplus and right side is tcpdump

The code is as follows:
```C++
#include "PcapFileDevice.h"
#include "Layer.h"
#include "TcpLayer.h"
#include "IPv4Layer.h"
#include "SSHLayer.h"
#include "Packet.h"
#include <iostream>
#include <fstream>
#include <stdlib.h>
#include <string.h>
#include <map>
#include <tuple>

int main(int argc, char* argv[])
{
    //if less arguments
    if(argc<3){std::cerr<<"Usage: "<<argv[0]<<" PcapFile BatchSize"<<std::endl; return -1;}
    //checking if pcap file exists
    std::ifstream f;
    f.open(argv[1]);
    if(!f){std::cerr<<"PcapFile doesnt Exist!!"<<std::endl; return -1;}
    f.close();
    // use the IFileReaderDevice interface to automatically identify file type (pcap/pcap-ng)
    // and create an interface instance that both readers implement
    pcpp::IFileReaderDevice* reader = pcpp::IFileReaderDevice::getReader(argv[1]);

    // verify that a reader interface was indeed created
    if (reader == NULL)
    {
        std::cerr << "Cannot determine reader for file type" << std::endl;
        return -1;
    }
    // open the reader for reading
    if (!reader->open())
    {
        std::cerr << "Cannot open "<<argv[1]<<" for reading" << std::endl;
        return -1;
    }
    if(strlen(argv[2]) >= 19){
        std::cerr<<"Too long number for Batch Size"<<std::endl;
        std::cerr<<"18 digits Maximum"<<std::endl;
        return -1;
    }
    char batch[18];
    char *end;
    strcpy(batch,argv[2]);
    pcpp::RawPacket rawPacket; //raw packet taken from pcap
    std::map<std::tuple<std::string,int,std::string>,std::tuple<time_t,unsigned long>> sshBFScore; // map of (src IP src port dest IP)as key, (timestamp and count of packets to it) as values
    unsigned long long counter = 0,batch_num;
    time_t current_time;
    batch_num = std::strtoull(batch,&end,10);
    while(reader->getNextPacket(rawPacket)){
        pcpp::Packet parsedPacket(&rawPacket); // making parsed packet from raw packet to extract information layer by layer
        if (parsedPacket.isPacketOfType(pcpp::IPv4)) // if the packet is of IPv4
        {
            if(!parsedPacket.isPacketOfType(pcpp::SSH)){continue;} // continue if the packet is not of ssh
            else{
                std::cout<<parsedPacket.toString()<<std::endl;
            }
        }
        //std::cout<<counter<<std::endl;
        if(counter == batch_num){
            //std::cout<<"invoked";
            // displaying all src IPs and rate they sent
        }
    }
}
```

**Comment by AlienX2001** [comment]
I tried filtering it with port instead of isPacketOf now i got more packets then what i was getting earlier. I set some counters and print statements and got that it doesnt give all the packets with isPacketOf, it only gives packets which have ssh layer on it.
Now I can see why would do that since there could be other services on port 22 as well. But in my case it isnt giving me packet as I am 100% sure they are all ssh traffic, since i am reading from a pcap which i made during an ssh brute force attack session.

The screenshot is as follows:
![image](https://user-images.githubusercontent.com/64488123/171051671-5fbbe4bd-e4e7-439c-bea1-a79796c925a8.png)
